docs/deployment/overview.md

```diff
@@ -1,47 +1,301 @@
+    ports:
+      - "8002:8001"
+      - "8004:8003"
+    depends_on:
+      pg:
+        condition: "service_started"
+    networks:
+      - "aspire"
+```
 
-With these commands, you can build your apps, containerize them, and deploy them to Kubernetes clusters. For more information, see [Aspirate (formerly Aspir8)](https://prom3theu5.github.io/aspirational-manifests/).
+Key points:
 
-### Manually create Kubernetes manifests
+- `${PG_PASSWORD}` and `${DBSETUP_IMAGE}` (and similar) are *placeholders* in the published asset.
+- They are not resolved during `aspire publish`.
+- A deployment engine (which could be `aspire deploy`, `docker compose` plus a script exporting variables, CI/CD variable injection, etc.) supplies their values later.
+- This keeps secrets and environment-specific values decoupled from generated structure.
 
-Alternatively, the Kubernetes manifests can be created manually. This involves more effort and is more time consuming. For more information, see [Deploy a .NET microservice to Kubernetes](/training/modules/dotnet-deploy-microservices-kubernetes/).
+Different integrations may use different placeholder conventions (environment variables, tokens, or parameter metadata), but the principle remains: publish preserves *shape*, deploy injects *values*.
+
+## Publisher model and compute environments
+
+.NET Aspire uses a flexible publisher model that distributes publishing behavior across your application graph. Resources support publishing and deployment through annotations:
+
+- `PublishingCallbackAnnotation` can be added to a resource to support `aspire publish`
+- `DeployingCallbackAnnotation` can be added to a resource to support `aspire deploy`
+
+This design enables hybrid and heterogeneous deployments, where different services within the same app can be deployed to different targets (cloud, edge, local).
+
+### Compute environments
+
+A **compute environment** is a core deployment concept in .NET Aspire that represents a target platform where your application resources will be deployed. Compute environments define how resources should be transformed and what deployment artifacts should be generated.
+
+**Compute resources** are the runnable parts of your application, such as .NET projects, containers, and executables that need to be deployed to a compute environment.
+
+When you add a compute environment like Docker Compose or Kubernetes, Aspire applies the correct publishing behavior to all compatible compute resources in your app model—no extra configuration needed.
+
+### Multiple environments require disambiguation
+
+If you add multiple compute environments, Aspire needs to know which resource goes where. Compute environments apply their transformations to all applicable compute resources (projects, containers, executables). If more than one environment matches a given resource, Aspire throws an ambiguous environment exception at publish time.
```

Comment:
I tried finding this exception that is mentioned, but could not find it in the Aspire repo. I want to confirm it's accurate.